spki/src/spki.rs

```diff
@@ -36,7 +39,7 @@ use der::pem::PemLabel;
 #[derive(Copy, Clone, Debug, Eq, PartialEq)]
 pub struct SubjectPublicKeyInfo<'a> {
     /// X.509 [`AlgorithmIdentifier`] for the public key type
-    pub algorithm: AlgorithmIdentifier<'a>,
+    pub algorithm: AlgorithmIdentifierRef<'a>,
```

Comment:
Note that this would be a lot more useful if `Params` were also added to `SubjectPublicKeyInfo`, i.e. this were changed to `AlgorithmIdentifier<Params>`.

It should probably get tackled as part of this PR.

Reply:
Likewise it should be changed on `PrivateKeyInfo` in the `pkcs8` crate

Reply:
I think it should be some type bound rather than generic `<Params>`

Reply:
After trying to implement generic arguments to SPKI or to Certificate/TBSCertificate, I think that `Params` might need to be bound with something like `Into<AnyRef>`, or any other generic enough type, so that there would be no need to add type parameters to these structures. I might be wrong with the SPKI, but handling type parameters for Certificate looks like a complete mess.

Reply:
I don't follow what you mean by "type bound" if not a generic parameter.

If you need additional bounds you can always specify them. But requiring one would be rather onerous I think, especially as the conversion for e.g. `ObjectIdentifier` only works for `&ObjectIdentifier`.

Reply:
I'm still playing here. I have the feeling that TBSCertitificate & company should not have generic parameters per se. Otherwise handling them easily becomes a mess. I'm still trying to fit SPKI / SPKIRef properly into my code. Maybe this will flow better once I update RSA crate to return specific SPKI rather than generic SPKIRef.

Reply:
Yes, they should use a concrete type. The only things I think it makes sense parameterizing are `AlgorithmIdentifier`, `SubjectPublicKeyInfo`, and `PrivateKeyInfo`.

Reply:
Is there any kind of performing `Into<AnyRef>` or `TryInto<AnyRef>` conversion? The only way I have at this moment is to encode the `AlgorithmIndeifier<Params>` into DER format and then reparse it again into `AlgorithmIndentifierRef`.